ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_utility.py

```diff
@@ -0,0 +1,51 @@
+# local
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_test
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.optional_get_element",
+    dtype_and_x=helpers.dtype_and_values(
+        dtype=[
+            "int8",
+            "int16",
+            "int32",
+            "int64",
+            "complex64",
+            "complex128",
+            "bool",
+            "float16",
+            "float32",
+            "float64",
+            "string",
+            "uint8",
+            "uint16",
+            "uint32",
+        ],
+        min_value=-100,
+        max_value=100,
+        min_num_dims=0,
+        max_num_dims=5,
+        min_dim_size=0,
+        max_dim_size=5,
+        allow_nan=True,
+    ),
+)
+def test_optional_get_element(
+    *,
+    dtype_and_x,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+):
+    input_dtype, x = dtype_and_x
+
+    helpers.test_function(
+        input_dtypes=input_dtype,
+        test_flags=test_flags,
+        on_device=on_device,
+        backend_to_test=backend_fw,
+        fn_name=fn_name,
+        x=x[0],
```

Comment:
You would need to test if the input is a list or a tensor. I would suggest if you drawn a `bool` which decides if you should input a list or a tensor.
```suggestion
        x=x[0] if input_tensor else x,
```

Reply:
> You would need to test if the input is a list or a tensor.

Sorry, I am not able to understand what you mean. Can you please explain?
And below is the output of what x is assigned to:
```
[array(0, dtype=int8)]
[array(-2.44758124e-308-2.44758124e-308j)]
[array([-3.33333333e-001-9.00719925e+015j,
        4.61104000e+016-2.22044605e-016j,
        1.40129846e-045+1.10000000e+000j,
       -3.62740835e-126-5.00000000e-001j,
       -1.00000000e+000+8.85831200e+281j,
        2.34160834e+016+1.00000000e-005j,
        3.40282347e+038-9.00719925e+015j,
        6.10351562e-005-4.78016237e+016j])]
[array(False)]
```